src/mds/schedule/scheduler_helper.cpp

```diff
@@ -379,6 +379,43 @@ void SchedulerHelper::CopySetDistributionInOnlineChunkServer(
         }
     }
 }
+
+void SchedulerHelper::CopySetDistributionInOnlineChunkServer(
+        const ChunkServerStatus status,
+        const std::vector<CopySetInfo> &copysetList,
+        const std::vector<ChunkServerInfo> &chunkserverList,
+        std::map<ChunkServerIdType, std::vector<CopySetInfo>> *out) {
+    // calculate the copysetlist by traversing the copyset list
+    for (auto item : copysetList) {
+        for (auto peer : item.peers) {
+            if (out->find(peer.id) == out->end()) {
+                (*out)[peer.id] = std::vector<CopySetInfo>{item};
+            } else {
+                (*out)[peer.id].emplace_back(item);
+            }
+        }
+    }
+
+    for (auto item : chunkserverList) {
+        // remove offline chunkserver
+        if (item.IsOffline()) {
+            out->erase(item.info.id);
+            continue;
+        }
+
+        if (item.status != status) {
```

Comment:
Find the chunkserver with the same status, this judgment condition is enough.

Reply:
done